### DISCUSSION TITLE
Interactive Matplotlib Plots using Widget are Overwritten

### DISCUSSION THREAD

**Discussion by ahunter1289** [gdsfactory/gdsfactory#2712] [Q&A] [answered]
In the below example, the first plot gets overwritten by the second plot. It has to do with using the widget interactive plots. I've tried several things. Any ideas how to resolve this?

`import gdsfactory as gf
import matplotlib.pyplot as plt
%matplotlib widget

c = gf.components.ring_double_pn(add_gap=0.3, drop_gap=0.3, radius=5.0, doping_angle=85, doped_heater=True, doped_heater_angle_buffer=10, doped_heater_layer='NPP', doped_heater_width=0.5, doped_heater_waveguide_offset=2.175)
c.plot('matplotlib')

d = gf.components.ring_double_pn(add_gap=0.3, drop_gap=0.3, radius=5.0, doping_angle=120, doped_heater=True, doped_heater_angle_buffer=10, doped_heater_layer='NPP', doped_heater_width=0.5, doped_heater_waveguide_offset=2.175)
d.plot('matplotlib')`

**Comment by ahunter1289** [comment]
I found that c.plot('kweb') resolves my problem. However, now I lose the capability to plot port numbers. Does anyone know how to plot port numbers on the plot when using 'kweb'?

**Comment by joamatab** [comment, marked as answer]
why not use klayout to zoom interactively?

c.plot() # plot fixed image
c.show() # show in klayout

**Reply by ahunter1289** [reply]
I haven't tried the klayout integration. i'll try that!

When i ran c.plot('kweb'), i got the prompt "You need to install kweb with `pip install 'gdsfactory[cad]'`". I then did a pip install gdsfactory[cad] and now when i use c.plot() with the %matplotlib widget everything is working how I hoped with interactive plotting but not conclusive that c.plot() didn't work before installed gdsfactory[cad]. c.plot('matplotlib') still has the overwrite issue but thats okay, i can use static plots when using c.plot('matplotlib') to look at port numbers, and then will try c.show() in klayout for zoom. 

Thank you!

**Reply by ahunter1289** [reply]
The answer to the original question is the following. The following is helpful as opposed to other plotting methods and visualizations because it shows ports clearly and you can zoom in and maintain view of the grid coordinate axis labels

%matplotlib widget
c.plot('matplotlib',new_window=True)
